apps/server/lib/lexical/server/boot.ex

```diff
@@ -77,10 +78,8 @@ defmodule Lexical.Server.Boot do
 
   defp load_app_modules(app_name) do
     Application.ensure_loaded(app_name)
-
-    with {:ok, modules} <- :application.get_key(app_name, :modules) do
-      Enum.each(modules, &Code.ensure_loaded!/1)
-    end
+    modules = Application.spec(app_name, :modules)
+    Code.ensure_all_loaded!(modules)
```

Comment:
This function is not available in Elixir < 1.15. 
The dream of using the elixir stdlib dies here.

```suggestion
  with {:ok, modules} <- :application.get_key(app_name, :modules),
          :ok <- :code.ensure_modules_loaded(modules) do 
         :ok 
  else _ -> 
    raise "Could not load modules in application #{app_name}" 
 end
 
```